op-dispute-mon/metrics/metrics.go

```diff
@@ -59,6 +59,9 @@ const (
 	AgreeDefenderAhead
 	DisagreeDefenderAhead
 
+	// L2 Block Number Challenge
+	DisagreeL2BlockChallenge
```

Comment:
I don't think this is a status we need.  It's either `AgreeChallengerAhead`, `DisagreeChallengerAhead`, `AgreeChallengerWins` or `DisagreeChallengerWins`.  The l2 block number challenge is a _way_ to win the game, but it's not in itself a new final status for the game.

Reply:
Right, so the intent here was to add another counter so we can as Mofi described "have the op-dispute-mon" yell loudly if it sees that the L2BlockChallenge has passed but we agree with the output and l2 block number for the game (shouldn't happen).

If preferred, can add a separate metric for this, though I don't mind this grouping since it's close to forecasting.